elasticsearch-transport/lib/elasticsearch/transport/client.rb

```diff
@@ -105,8 +105,8 @@ class Client
       # @yield [faraday] Access and configure the `Faraday::Connection` instance directly with a block
       #
       def initialize(arguments={}, &block)
-        @options = arguments
-        @arguments = arguments
+        @options = arguments.inject({}){|args,(k,v)| args[k.to_sym] = v; args}
```

Comment:
nit: `each_with_object` gives you the return value of the memo_obj so you don't have to add the `; args` at the end of your block.

```suggestion
        @options = arguments.each_with_object({}){ |(k,v), args| args[k.to_sym] = v }
```